### PR TITLE
DROOLS-5808 : XLS to GDST conversion should ensure types work

### PR DESCRIPTION
**Pull request by Rikkola**
DROOLS-5807 : GDST to XLS conversion should not convert dialect column

Moving few classes from drools-wb so they can be used by both GDST->XLS and XLS-GDST conversion.

**JIRA**: [DROOLS-5807](https://issues.redhat.com/browse/DROOLS-5807)
**JIRA**: [DROOLS-5808](https://issues.redhat.com/browse/DROOLS-5808)

 
**Referenced Pull Requests**:
* https://github.com/kiegroup/kie-wb-common/pull/3483
* https://github.com/kiegroup/drools-wb/pull/1441

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java [hunk 1]
```diff
@@ -144,7 +144,7 @@ private GuidedDecisionTable52 makeTempLHSGuidedDecisionTable(final BRLConditionC
         for (int i = 0; i < childColumns.size(); i++) {
             final BRLVariableColumn brlConditionVariableColumn = childColumns.get(i);
             columnContext.addBoundName(brlConditionVariableColumn.getVarName());
-            if (brlConditionVariableColumn.getVarName().isEmpty()) {
+            if (brlConditionVariableColumn.getVarName() == null || brlConditionVariableColumn.getVarName().isEmpty()) {
```
Comment (Rikkola) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java:147]: 3rd issue the file found, but I did not make an ticket for it. BRL columns with no variables caused an NPE here.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnDataBuilderDefault.java [hunk 2]
```diff
@@ -121,7 +121,8 @@ private String build() {
 
         private boolean isMissingVariable(final BRLColumn brlColumn) {
             for (final Object childColumn : brlColumn.getChildColumns()) {
-                if (!getVariableName(childColumn).isEmpty()) {
+                final String variableName = getVariableName(childColumn);
+                if (variableName != null && !variableName.isEmpty()) {
                     return false;
```
Comment (vpellegrino) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnDataBuilderDefault.java:126]:
```suggestion
                final String variableName = getVariableName(childColumn);
                return StringUtils.isEmpty(variableName);
```

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java (hunk 1 above)
Comment (vpellegrino) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java:147]: Also there. I think that the usage of `StringUtils.isEmpty()` whould make statement more readable.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/util/Skipper.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+public class Skipper {
+
+    private Skipper() {
+    }
+
+    public static boolean shouldSkip(final NotificationReporter notificationReporter,
+                                     final BaseColumn baseColumn) {
+        if (baseColumn instanceof AttributeCol52) {
+            notificationReporter.report("Dialect is not a supported column type in XLS Decision tables. Conversion ignored this column.");
```
Comment (vpellegrino) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/util/Skipper.java:31]: this string is not translated by any i18n service?

**Review by vpellegrino — CHANGES_REQUESTED** [review]
I've added minor comments. Overall code looks clear to me

**Comment by Rikkola** [comment]
Not sure how I mixed these, but these are the wrong conversion fixes. Closing the PRs for now.

**Pull request closed without merging**